aten/src/ATen/native/SummaryOps.cpp

```diff
@@ -65,4 +134,112 @@ _bincount_cpu(const Tensor& self, const Tensor& weights, int64_t minlength) {
   });
 }
 
+
+std::tuple<Tensor,Tensor> _histogram_cpu(
+    const Tensor& self,
+    int64_t bins,
+    c10::optional<ArrayRef<double>> range,
+    const Tensor& weights,
+    bool density) {
+
+  // Weights having a different shape from input is not supported yet. TO DO:
```

Comment:
This comment is pretty good but it's OK for the weights to have a different shape. It just has to have the same number of elements as self.

Reply:
If I'm looking at the code correctly, this checks the dimensions of the self and weights tensors, and if the dimensions are the same, it calls std::equal on the arrays of sizes.

Per the above comment: removing the redundant check in the next commit.

Reply:
You're completely right that the check is correct. My mistake.